sync/src/main/java/tech/pegasys/teku/sync/PeerSyncBlockRequest.java

```diff
@@ -37,7 +37,15 @@ public PeerSyncBlockRequest(
 
   @Override
   public SafeFuture<?> onResponse(final SignedBeaconBlock response) {
-    slotOfLastBlock = Optional.of(response.getSlot());
+    UnsignedLong newBlockSlot = response.getSlot();
+    if (maybeSlotOfLastBlock.isPresent()) {
+      UnsignedLong slotOfLastBlock = maybeSlotOfLastBlock.get();
+      if (newBlockSlot.compareTo(slotOfLastBlock) <= 0) {
```

Comment:
Would be nice to move these validations into `Eth2Peer` if possible so that the application logic doesn't need to worry about low-level validations that should apply for all blocksByRange requests.

Reply:
Done.